src/@types/pkg.ts

```diff
@@ -84,17 +84,17 @@ export interface RawPkg {
   humanDependents: string;
   changelogFilename: string | null;
   lastCrawl: string;
+  _revision: number;
   _searchInternal: {
-    expiresAt: number;
     alternativeNames: string[];
     popularAlternativeNames: string[];
   };
 }
 
 export type FinalPkg = RawPkg & {
-  _searchInternal: {
-    downloadsMagnitude?: number;
-    jsDelivrPopularity?: number;
-    popularName?: string;
-  };
+  _oneTimeDataToUpdateAt?: number;
+  _periodicDataUpdatedAt?: number;
+  _jsDelivrPopularity?: number;
+  _downloadsMagnitude?: number;
+  _popularName?: string;
```

Comment:
I know, but this was necessary because partial updates wouldn't work with nested fields, so I needed to pull them to the top level. All of these fields have more useful "public" equivalents, so I do hope nobody used the internal ones.

Reply:
We can define some [unretrievableAttributes](https://www.algolia.com/doc/api-reference/api-parameters/unretrievableAttributes/) at the index level so those fields would not be returned by the search requests, but only used internally for the ranking/queuing system

Reply:
I already declared the queue-related attributes as unretrievable. The question is `_jsDelivrPopularity` and `_downloadsMagnitude` which were previously in a "private" but retrievable `_searchInternal` object and needed to be moved to allow partial updating.

Reply:
Im not too worried but it could be fair to create a v2 index and reach out to current customer

Reply:
It's an option, but considering the current one is already quite broken (not only packages are missing but also all jsDelivr hits are zero there, etc.) I think it makes more sense to push this out for everyone.